README.md

```diff
@@ -1,2 +1,257 @@
+
+We at [Astea Solutions](https://asteasolutions.com/) made this library because of the duplication of work when creating a documentation for an API that uses `zod` to validate request input and output.
+
+## Usage
+
+### Installation
+
+```shell
+yarn add @asteasolutions/zod-to-openapi
+# or
+npm install @asteasolutions/zod-to-openapi
+```
+
+### Expanding the zod functionalities
+
+In order to specify some OpenAPI specific metadata you should use the exported `extendZodWithOpenApi`
+function with your own instance of `zod`.
+
+Note: This should be done only once in a common-entrypoint file of your project (for example an `index.ts`/`app.ts`)
+
+```ts
+import { extendZodWithOpenApi } from '@asteasolutions/@asteasolution/zod-to-openapi';
+import { z } from 'zod';
+
+extendZodWithOpenApi(z);
+
+// We can now use `.openapi()` to specify OpenAPI metadata
+z.string().openapi({ description: 'Some string' });
+```
+
+### Generating components
+
+The `OpenAPIRegistry` class is used as a utility for creating definitions that are then to be used to
+generate the OpenAPI document using the `OpenAPIGenerator` class. In order to generate components the `generateComponents` method should be used.
+
+```ts
+import {
+  OpenAPIRegistry,
+  OpenAPIGenerator,
+} from '@asteasolutions/@asteasolution/zod-to-openapi';
+
+const registry = new OpenAPIRegistry();
+
+// Register definitions here
+
+const generator = new OpenAPIGenerator(registry.definitions);
+
+return generator.generateComponents();
+```
+
+### Registering schema definitions
+
+An OpenAPI schema should be registered using the `register` method of an `OpenAPIRegistry` instance.
+
+```ts
+const UserSchema = registry.register(
+  'User',
+  z.object({
+    id: z.string().openapi({
+      example: '1212121',
+    }),
+    name: z.string().openapi({
+      example: 'John Doe',
+    }),
+    age: z.number().openapi({
+      example: 42,
+    }),
+  })
+);
+```
+
+The YAML equivalent of the schema above would be:
+
+```yaml
+User:
+  type: object
+  properties:
+    id:
+      type: string
+      example: '1212121'
+    name:
+      type: string
+      example: John Doe
+    age:
+      type: number
+      example: 42
+  required:
+    - id
+    - name
+    - age
+```
+
+Note: All properties defined inside `.openapi` of a single zod schema are applied at their appropriate schema level.
+
+The result would be an object like `{ components: { schemas: { User: {...} } } }`. The key for the object is the value of the first argument passed to `.register` (in this case - `User`).
+
+The resulting schema can then be referenced by using `$ref: #/components/schemas/User` in an existing OpenAPI JSON.
+
+### Registering parameter definitions
+
+An OpenAPI parameter (query/path/header) should be registered using the `registerParameter` method of an `OpenAPIRegistry` instance.
+
+```ts
+const UserIdSchema = registry.registerParameter(
+  'UserId',
+  z.string().openapi({
+    param: {
+      name: 'id',
+      in: 'path',
```

Comment:
Maybe `path` can be `example-path`

Reply:
The `in` key is openapi specific - `'query' | 'header' | 'path' | 'cookie'`